api/pkg/apis/v1alpha1/providers/stage/wait/wait.go

```diff
@@ -301,6 +301,23 @@ func (i *WaitStageProvider) Process(ctx context.Context, mgrContext contexts.Man
 			return outputs, false, nil
 		}
 		counter++
+		if counter%10 == 0 {
+			// Avoid to check the activation status too frequently
+			activationName, ok := inputs["__activation"].(string)
+			if !ok || activationName == "" {
+				log.InfoCtx(ctx, "  P (Wait Processor): __activation is not provided, exiting endless wait.")
+				return nil, false, err
+			}
+			_, err := i.ApiClient.GetActivation(ctx, activationName, namespace, i.Config.User, i.Config.Password)
+			if err != nil && strings.Contains(err.Error(), v1alpha2.NotFound.String()) {
+				// Since we use ApiClient to get the activation, not found error will become v1alpha2.InternalError
+				// format: utils.SummarySpecError{Code:\"Symphony API: [500]\", Message:\"Not Found: ..."}
+				// We need to check if it contains v1alpha2.NotFound
+				log.InfoCtx(ctx, "  P (Wait Processor): detected activation got deleted, exiting endless wait.")
+				return nil, false, err
```

Comment:
Given activation is deleted in this case, if we return err to stage-manager, will this cause problem?

Reply:
In this case, the `providers.stage.wait` fails since we can't get all required objects. The stage manager will receive the error shows that `providers.stage.wait` quits since the activation does not exist any more. I refined the error type in the new version.